### PR TITLE
[WIP] remove : dependency on u() completely

### PR DESCRIPTION
**Pull request by kritkaran94**
WIP to Remove u() function completely from compatibility

### CODE REVIEW

**Comment by kritkaran94** [comment]
@jksuom This PR addresses Issue #10985. I am getting an assertion error in ``test_pretty.py`` and I am stuck on it. 

Can you review this please ?

````

________________________________________________________________________________
_______ sympy/printing/pretty/tests/test_pretty.py:test_pretty_piecewise _______
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sympy-1.0.1.dev0-py2.7.egg/sympy/printing/pretty/tests/test_pretty.py", line 2969, in test_pretty_piecewise
    assert upretty(expr) == ucode_str
AssertionError
________________________________________________________________________________
____________ sympy/printing/pretty/tests/test_pretty.py:test_hyper _____________
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sympy-1.0.1.dev0-py2.7.egg/sympy/printing/pretty/tests/test_pretty.py", line 4507, in test_hyper
    assert upretty(expr) == ucode_str
AssertionError
________________________________________________________________________________
___________ sympy/printing/pretty/tests/test_pretty.py:test_meijerg ____________
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sympy-1.0.1.dev0-py2.7.egg/sympy/printing/pretty/tests/test_pretty.py", line 4627, in test_meijerg
    assert upretty(expr) == ucode_str
AssertionError
 tests finished: 1441 passed, 3 failed, 17 skipped, 13 expected to fail, 
1 expected to fail but passed, in 208.94 seconds 
DO *NOT* COMMIT!

````

**Comment by jksuom** [comment]
I'll try to look into it. 

BTW, you had ca. 20 builds in queue, many of which test the deletion of one line only. It is not necessary to push each one of them separately to github. You could first test them locally, and then push several commits together. I have cancelled most of the builds to save time.

**Comment by kritkaran94** [comment]
@jksuom I am sorry I messed up that. Would keep in mind from next time.
Please review when you get time.

**Comment by jksuom** [comment]
It seems that the strings in [`ucode_pics`](https://github.com/sympy/sympy/blob/master/sympy/printing/pretty/pretty_symbology.py#L566) should be rewritten as unicode strings when the function `u()` is removed. If I am not mistaken, the string in 'F' should become

`u'\u250c\u2500\n\u251c\u2500\n\u2575'`

instead of the original
```
u('\N{BOX DRAWINGS LIGHT DOWN AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\n'
                            '\N{BOX DRAWINGS LIGHT VERTICAL AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\n'
                            '\N{BOX DRAWINGS LIGHT UP}')
```

and similarly the string in 'G' should be

`u'\u256d\u2500\u256e\n\u2502\u2576\u2510\n\u2570\u2500\u256f'`

instead of
```
u('\N{BOX DRAWINGS LIGHT ARC DOWN AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\N{BOX DRAWINGS LIGHT ARC DOWN AND LEFT}\n'
                '\N{BOX DRAWINGS LIGHT VERTICAL}\N{BOX DRAWINGS LIGHT RIGHT}\N{BOX DRAWINGS LIGHT DOWN AND LEFT}\n'
                '\N{BOX DRAWINGS LIGHT ARC UP AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\N{BOX DRAWINGS LIGHT ARC UP AND LEFT}')
```

**Comment by kritkaran94** [comment]
I have made the changes. 
I feel you are absolutely right and unicode strings would be required for the same.(``pretty_symbology.py``)

**Comment by kritkaran94** [comment]
@jksuom There are few issues with removal of ``u`` in the following files :

a.) ``sympy/vector/dyadic.py`` 
b.) ``sympy/vector/vector.py``
c.) ``sympy/vector/scaler.py`

In [dyadic.py](https://github.com/sympy/sympy/blob/master/sympy/vector/dyadic.py#L200)
````
obj._pretty_form = u('(' + vector1._pretty_form + '|' +
                             vector2._pretty_form + ')')
obj._latex_form = ('(' + vector1._latex_form + "{|}" +
                           vector2._latex_form + ')')
````
In [vector.py](https://github.com/sympy/sympy/blob/master/sympy/vector/vector.py#L368)
````
obj._pretty_form = u(pretty_str)
````
In [scaler.py](https://github.com/sympy/sympy/blob/master/sympy/vector/scalar.py#L37)
````
obj._pretty_form = u(pretty_str)
````

How would these be dealt with if u() is removed ?

**Comment by jksuom** [comment]
I'm not sure how this should be solved. I would start by testing if `u(pretty_str)` could be replaced with `u'' + pretty_str`. If that works, then I'd similarly change `u('('` to `u'('` and drop the final `)` in the `obj._pretty_form` line of `dyadic.py`.

File: sympy/physics/quantum/hilbert.py [hunk 1]
```diff
@@ -528,7 +528,7 @@ def _pretty(self, printer, *args):
             pform = prettyForm(*pform.right(next_pform))
             if i != length - 1:
                 if printer._use_unicode:
-                    pform = prettyForm(*pform.right(u(' ') + u('\N{CIRCLED PLUS}') + u(' ')))
+                    pform = prettyForm(*pform.right(u' ' + u'\N{CIRCLED PLUS}' + u' '))
```
Comment (asmeurer) [inline sympy/physics/quantum/hilbert.py]: I think these can be made into a single string, like `u' \N{CIRCLED PLUS} '`.

File: sympy/printing/pretty/pretty.py [hunk 2]
```diff
@@ -1734,7 +1734,7 @@ def _print_PolyRing(self, ring):
     def _print_FracField(self, field):
         return prettyForm(sstr(field))
 
-    def _print_FreeGroupElm(self, elm):
+    def _print_FreeGroupElement(self, elm):
```
Comment (asmeurer) [inline sympy/printing/pretty/pretty.py]: Why is this change here?

File: sympy/printing/pretty/pretty_symbology.py [hunk 3]
```diff
@@ -564,13 +564,8 @@ def annotated(letter):
...
     ucode_pics = {
-        'F': (2, 0, 2, 0, u('\N{BOX DRAWINGS LIGHT DOWN AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\n'
-                            '\N{BOX DRAWINGS LIGHT VERTICAL AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\n'
-                            '\N{BOX DRAWINGS LIGHT UP}')),
-        'G': (3, 0, 3, 1,
-              u('\N{BOX DRAWINGS LIGHT ARC DOWN AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\N{BOX DRAWINGS LIGHT ARC DOWN AND LEFT}\n'
-                '\N{BOX DRAWINGS LIGHT VERTICAL}\N{BOX DRAWINGS LIGHT RIGHT}\N{BOX DRAWINGS LIGHT DOWN AND LEFT}\n'
-                '\N{BOX DRAWINGS LIGHT ARC UP AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\N{BOX DRAWINGS LIGHT ARC UP AND LEFT}'))
+        'F': (2, 0, 2, 0, u'\u250c\u2500\n\u251c\u2500\n\u2575'),
+        'G': (3, 0, 3, 1, u'\u256d\u2500\u256e\n\u2502\u2576\u2510\n\u2570\u2500\u256f')
```
Comment (asmeurer) [inline sympy/printing/pretty/pretty_symbology.py]: Please keep this in the original verbose form.

**Comment by asmeurer** [comment]
I assume you've checked that `u` was unused in all the files that you removed it from.

**Comment by asmeurer** [comment]
`u(variable)` should be something like `variable.decode('utf-8') if PY2 else variable`.

**Comment by asmeurer** [comment]
Unless `variable` is actually `bytes` in Python 3, in which case just `variable.decode('utf-8')`.

File: sympy/printing/pretty/pretty_symbology.py (hunk 3 above)
Comment (jksuom) [inline sympy/printing/pretty/pretty_symbology.py]: Python 2 does not recognize the verbose form. It currently applies `codecs.unicode_escape_decode(x)[0]` which is how `u(x)` is defined in `compatibility`. The decoded form seems necessary if `u` is to be removed. Maybe the verbose form could be preserved as a comment and brought back when Python 2 is no more supported.

Comment (asmeurer) [inline sympy/printing/pretty/pretty_symbology.py]: So perhaps we should keep `u` for these specific cases. Or else just call `codecs.unicode_escape_decode(x)[0]` manually.

Comment (jksuom) [inline sympy/printing/pretty/pretty_symbology.py]: It seems to me then that `u` should be preserved in `compatibility` for now and only called in a few special places.

File: sympy/physics/quantum/hilbert.py (hunk 1 above)
Comment (kritkaran94) [inline sympy/physics/quantum/hilbert.py]: Okay, I will try that.

File: sympy/printing/pretty/pretty.py (hunk 2 above)
Comment (kritkaran94) [inline sympy/printing/pretty/pretty.py]: I will correct that.

**Comment by kritkaran94** [comment]
@asmeurer Yes, I have checked properly that ``u`` was unused in the files from which it has been removed.

File: sympy/printing/pretty/pretty_symbology.py (hunk 3 above)
Comment (kritkaran94) [inline sympy/printing/pretty/pretty_symbology.py]: I too agree that ``u`` should be called in a few special cases. I will point out the set of files which must have ``u`` (special cases) after going through the remaining ones:

As of now after the above conversation I think 

a.) ``sympy/printing/pretty/pretty_symbology.py``
b.) ``sympy/vector/dyadic.py ``
c.) ``sympy/vector/vector.py``
d.) ``sympy/vector/scaler.py``

Should be the ones which must use ``u`` from ``sympy.compatibility``.

**Comment by kritkaran94** [comment]
There a few more test files which don't require ``u`` to be imported.
@jksuom Do you feel we could do better with this PR or does it look good ?

**Comment by jksuom** [comment]
Did you mean that some imports of `u` remain in addition to those in the four files a - d?

**Comment by kritkaran94** [comment]
Yes I meant that. For instance ``sympy/core/tests/test_numbers.py``

**Comment by jksuom** [comment]
Perhaps you could find the remaining ones as well and push them all at the same time.

**Comment by kritkaran94** [comment]
There were only two more files 

a. ``test_matrices.py``
b. ``test_numbers.py``

So pushed one after the other (would ensure local testing from next PR)

Does it look good ?

**Comment by jksuom** [comment]
This is what I see: test_containers and test_matrices have import `u`. You could test if they can be removed.

**Comment by jksuom** [comment]
It would probably be also useful to record the remaining imports. Maybe a couple of lines could be added to `compatibility`.

**Comment by kritkaran94** [comment]
I will test ``test_containers.py``.

Did you mean writing a docstring in ``compatibility.py`` describing the
files which still have ``u`` imported?

>

**Comment by kritkaran94** [comment]
I will test ``test_containers.py``.

Did you mean writing a docstring in ``compatibility.py`` describing the
files which still have ``u`` imported?

**Comment by jksuom** [comment]
I think that could be a suitable place. When Python 2 will no longer be supported, it will be easy to find those imports.

**Comment by asmeurer** [comment]
Can you add a note in the compatibility file that it's needed to handle \N.

**Comment by kritkaran94** [comment]
I will add notes in ``compatibility.py`` regarding the need to handle ``\N`` and the names of the files too by today and get it reviewed.

**Comment by jksuom** [comment]
I have continued testing with Python 2.7. It looks like it may not be necessary to import u (and codecs) after all since it seems possible to decode unicode escape strings using the string `.decode`.
```
In [6]: s = ('\N{BOX DRAWINGS LIGHT DOWN AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\n'
                            '\N{BOX DRAWINGS LIGHT VERTICAL AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\n'
                            '\N{BOX DRAWINGS LIGHT UP}')

In [7]: from sympy.core.compatibility import u

In [8]: u(s)
Out[8]: u'\u250c\u2500\n\u251c\u2500\n\u2575'

In [9]: s.decode('unicode-escape')
Out[9]: u'\u250c\u2500\n\u251c\u2500\n\u2575'
```
The same decode method would probably work also in the other cases.

**Comment by kritkaran94** [comment]
Ping @asmeurer.
Please give your take on the last comment by @jksuom so that the changes can be made accordingly.

**Comment by asmeurer** [comment]
@jksuom that can work, but the original string needs to be a byte string for it to work in Python 3 (like `b'\N{BOX DRAWINGS LIGHT DOWN AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\n\N{BOX DRAWINGS LIGHT VERTICAL AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\n\N{BOX DRAWINGS LIGHT UP}'`). This will work in Python 2 as well because `b'a'` is the same as `'a'` in Python 2.

**Comment by kritkaran94** [comment]
@jksuom I have added a comment for the `\N` requirement and the names of the files in which `u` is still required. 

Please let me know about any other changes.

**Comment by kritkaran94** [comment]
The tests have now passed can you please review this ?

**Comment by jksuom** [comment]
I'd prefer to have `u` removed altogether if possible. It looks like that could be done in `sympy/printing/pretty/pretty_symbology.py` by prepending the unicode escape strings by `b` to make them `bytes` in Python 3 (and to have no effect in Python 2), and then appending `.decode('unicode-escape')` to the whole sequence of strings.

**Comment by jksuom** [comment]
It is harder to say what should be done with vectors etc. It is conceivable that `u` would be needed in cases like this:
```
>>> A = CoordSysCartesian('\N{LATIN CAPITAL LETTER A WITH DIAERESIS}')
>>> A.i
\N{LATIN CAPITAL LETTER A WITH DIAERESIS}.i
>>> pretty(A.i)
u'\xc4_i'
```
but I doubt if anybody would want to use such constructions. The result does not look much better with utf-8 characters:
```
>>> A = CoordSysCartesian('Ä')
>>> A.i
Ä.i
>>> pretty(A.i)
u'\xc3\x84_i'
```
Hence I would consider it possible to omit the `u` in those files.

@asmeurer ping.

**Comment by asmeurer** [comment]
To make things simplest, I would always decode strings when they are created. It's not useful to have a byte string with the `\N`, except for readability of the source code.

**Comment by kritkaran94** [comment]
So, I guess for now the present changes in this PR are sufficient ?.

Ping @jksuom (``compatibility.py``) please give your views on the same.

**Comment by jksuom** [comment]
It looks like `u()` should, in fact, be left unchanged in those cases where it is applied to a variable string. But that is probably not necessary in `pretty_symbology`. It looks like adding the unicode prefix `u` in front of each one of the escape strings, not just the first one, should work also in Python 2.7:
```
u'\N{BOX DRAWINGS LIGHT DOWN AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\n'
    u'\N{BOX DRAWINGS LIGHT VERTICAL AND RIGHT}\N{BOX DRAWINGS LIGHT HORIZONTAL}\n'
    u'\N{BOX DRAWINGS LIGHT UP}'
```
Perhaps you could test this. (And change the lines added to `compatibility`, if it works.)

**Comment by asmeurer** [comment]
I would remove u entirely. It should be up to the user of the vector module to pass a properly decoded string (preferably a `unicode` string in Python 2 if it contains non-ascii). We can document this as well.

**Comment by jksuom** [comment]
I think that should be another PR. It seems that some functions in the vector module do not accept unicode strings when running under Python 2.7. Therefore more work is necessary.

**Comment by asmeurer** [comment]
Doesn't the `u` function return a unicode string?

**Comment by jksuom** [comment]
It does and it works well with the hard coded strings. But the functions apply `str` to input strings and that will accept only ascii strings.
```
In [2]: str(u'\xd6')
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
/home/jks/sympy/<ipython-input-2-2db82e2273a6> in <module>()
----> 1 str(u'\xd6')

UnicodeEncodeError: 'ascii' codec can't encode character u'\xd6' in position 0: ordinal not in range(128)
```

**Comment by asmeurer** [comment]
But does it work in master? From what I can tell, the `_pretty_form` is already saved as a unicode string (because `u` is used on `pretty_form`), so I see no difference from the current behavior than from requiring the user to explicitly pass in `pretty_form` as a `unicode` string.

**Comment by jksuom** [comment]
The arguments strings must be ascii in Python 2.7.  Even this
```
In [5]: A = CoordSysCartesian(u'\N{LATIN CAPITAL LETTER A WITH DIAERESIS}')
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
/home/jks/sympy/<ipython-input-5-5c151e0a1bc8> in <module>()
----> 1 A = CoordSysCartesian(u'\N{LATIN CAPITAL LETTER A WITH DIAERESIS}')

/home/jks/sympy/sympy/vector/coordsysrect.pyc in __new__(cls, name, location, rotation_matrix, parent, vector_names, variable_names)
     46         """
     47 
---> 48         name = str(name)
     49         Vector = sympy.vector.Vector
     50         BaseVector = sympy.vector.BaseVector

UnicodeEncodeError: 'ascii' codec can't encode character u'\xc4' in position 0: ordinal not in range(128)
```
does not work.

**Comment by asmeurer** [comment]
From what I see, only the pretty form is enforced to be unicode right now. Is the coordinate system name ever created from the pretty form?  If so, I don't see where, but the vector code is somewhat convoluted so it's easy to miss. 

If not, I wouldn't worry about non-ascii strings for anything other than the pretty_form (at least for Python 2), unless someone explicitly asks for it. `CoordSysCartesian(u'\N{LATIN CAPITAL LETTER A WITH DIAERESIS}')` already doesn't work in master, so it won't be a regression in this PR. For pretty_form itself, we just need to document that Python 2 users should input it as a decoded unicode string (and of course make sure that it actually works with unicode strings, but it ought to, since it is already unicode in master thanks to `u`).

**Comment by jksuom** [comment]
If removing those applications of `str` does no harm, then the `u`s could also be removed. But they are there probably for some reason.

**Comment by asmeurer** [comment]
The `str` is likely there so that you can input the name as a Symbol.

**Comment by jksuom** [comment]
I now see that it may be possible to construct unicode pretty forms for the coordinate vectors even if the name of the coordinate system is an ordinary string simply by appending the indices `_i` etc. as unicode strings.
```
In [9]: 'A' u'_i'
Out[9]: u'A_i'
```
So the solution could be this modification of `coordsysrect`.

**Comment by jksuom** [comment]
The compatibility function `u()` currently has a dual role:
- It decodes unicode escape strings containing `\N{}`.
- It transforms a `str` object to a `unicode` object.

It is the latter property that is primarily made use of in the `vector` module. It seems that it can be emulated in Python 2.7 simply by concatenating with `u''` (the empty unicode string) *provided that the string object consists of ascii characters only*:
```
>>> u'' + 'abc'
u'abc'
>>> u'' + 'äö'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 0: ordinal not in range(128)
```
Hence it seems possible to remove `u(pretty_str)` from `vector.py` and `scalar.py` by writing `u'' + pretty_str` instead. In `dyadic.py`,  the expression `u('(' + ...)` can be replaced by `(u'(' + ...)`.

This will have the following effect in Python 2.7.
- The `\N{}` escape is no longer supported:
```
>>> A = CoordSysCartesian('\N{LATIN CAPITAL LETTER A WITH DIAERESIS}')
>>> pprint(A.i)
\N{LATIN CAPITAL LETTER A WITH DIAERESIS}_i
```
while currently
```
>>> A = CoordSysCartesian('\N{LATIN CAPITAL LETTER A WITH DIAERESIS}')
>>> pprint(A.i)
Ä_i
```
- Non-ascii names raise an exception:
```
>>> A = CoordSysCartesian('Ä')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/vector/coordsysrect.py", line 120, in __new__
    pretty_vects[0], latex_vects[0])
  File "sympy/vector/vector.py", line 368, in __new__
    obj._pretty_form = u'' + pretty_str
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 0: ordinal not in range(128)
```
instead of the wrong rendering presented by the current master
```
>>> A = CoordSysCartesian('Ä')
>>> pprint(A.i)
Ã_i
```

**Comment by asmeurer** [comment]
In Python 2, non-u-prefixed string literals with non-ascii characters are basically undefined. They generally work if you decode them with utf-8, but that depends on the locale and the encoding of the document. 

Also, the `\N` escape in non-u-prefixed strings is also undefined. 

So my suggestion is to just document this, that if a user is in Python 2 and they want to use non-ASCII characters, they need to use unicode literals. And we need to make the code work with unicode strings as well, but for now, that's really only the case for the pretty printing strings (and maybe the name, but if you want a Unicode character in the name, you probably just want it for printing, so you can put it in the pretty_form instead). 

In Python 3, everything just works, because all strings are unicode strings.

**Comment by jksuom** [comment]
>  if a user is in Python 2 and they want to use non-ASCII characters, they need to use unicode literals.

It seems that unicode strings are currently accepted by `CoordSysCartesian` in Python 2 only if they consist of characters < 128. Should this be fixed before trying to remove `u()`? Or should we just accept that only ascii names can be given? (That should be no big issue.)

**Comment by kritkaran94** [comment]
@jksuom I am a bit confused here should I remove u from `dyadic.py`, `vector.py`, `scaler.py` or not? 
I guess it's doable as you had mentioned earlier.`

@asmeurer You want the below to be documented in `compatibility.py` right?
````
If a user is in Python 2 and wants to use non-ASCII characters, they need to use unicode literals.
However in Python 3, everything just works, because all strings are unicode strings.
````

**Comment by asmeurer** [comment]
I would document it specifically in the vector class (and any other class that takes a pretty_form), that unicode pretty forms in Python 2 should use the `u` prefix.

**Comment by kritkaran94** [comment]
Okay I will do that.

**Comment by kritkaran94** [comment]
@jksuom I have made the changes please review them.

**Comment by jksuom** [comment]
I think this is probably what should be done after `u()` is removed. The text in `compatibility` should record these changes so that they can be removed when Python 2.7 is no longer supported.

**Comment by kritkaran94** [comment]
@jksuom  So, we shouldn't change these files as of now ?

**Comment by jksuom** [comment]
This change is how I'd prefer to have the removal of `u()` done. Some functionality will be lost in Python 2.7: the use of `\N` encoded unicode names for coordinate systems, but I don't think that would matter too much.

I only wanted to point out that the text in `compatibility` should say something about the use of the unicode string prefix `u`. It is needed for Python 2.7 only, in Python >= 3.3 it is ignored and should probably be completely removed when 2.7 is no longer supported. Therefore it is useful to record this use in `compatibility`.

**Comment by kritkaran94** [comment]
@jksuom At this point of time, through this PR dependency on `u` has been completely removed, a note has been added in ``compatibility.py`` regarding the need to prefix unicode ``u`` in PY 2.7.

If it looks good to you please merge. Also, if there are any other changes with regards to the notes please let me know.

**Comment by jksuom** [comment]
I think this is good to merge unless some documentation should be added. I'll leave that for @asmeurer to decide.

**Comment by kritkaran94** [comment]
@asmeurer Please give your comments on this PR with regards to what else can be documented further.

**Comment by kritkaran94** [comment]
@asmeurer is there anything else to be documented. Please give your comments on the same.

File: sympy/core/compatibility.py [hunk 4]
```diff
@@ -59,6 +59,12 @@ class Foo(with_metaclass(Meta)):
         * Define class `Foo` with metaclass `Meta` and parent class `Bar`:
             class Foo(with_metaclass(Meta, Bar)):
                 pass
+Issue #10985:
```
Comment (asmeurer) [inline sympy/core/compatibility.py]: This comment is not needed, since you've removed `u` entirely.

File: sympy/vector/vector.py [hunk 5]
```diff
@@ -345,6 +344,9 @@ def separate(self):
 class BaseVector(Vector, AtomicExpr):
     """
     Class to denote a base vector.
+
+    Unicode pretty forms in Python 2 should use the `u` prefix.
```
Comment (asmeurer) [inline sympy/vector/vector.py]: You need two \`\` for code in docstrings, like \`\`u\`\`

**Comment by asmeurer** [comment]
As far as documentation, I think the comment you've got now in the docstring is fine.

File: sympy/vector/vector.py (hunk 5 above)
Comment (kritkaran94) [inline sympy/vector/vector.py]: @asmeurer Both the changes have been made. I guess this can be merged as soon as the tests pass.

**Comment by asmeurer** [comment]
+1

**Comment by kritkaran94** [comment]
@jksuom Please merge this @asmeurer has given a +1 for the same.

**Comment by jksuom** [comment]
Thanks, merging.
